runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java

```diff
@@ -193,15 +199,18 @@ public void onVarbitChanged(VarbitChanged event)
 
 	private void setOverlayRound(Role role)
 	{
-		// Prevent changing roles when a role is already set, as widgets can be
-		// loaded multiple times in game from eg. opening and closing the horn
-		// of glory.
-		if (overlay.getCurrentRound() != null)
+		// Check for null to prevent changing roles when a role is already set,
+		// as widgets can be loaded multiple times in game from eg. opening and
+		// closing the horn of glory.
+		if (timerOverlay.getCurrentRound() == null)
```

Comment:
Since there are multiple overlays now it may be worth storing the `Round` on the plugin and having the overlays access that.

Reply:
I was contemplating this. I can work on implementing it this way.